usr/src/uts/i86pc/os/intr.c

```diff
@@ -471,6 +472,7 @@
 #include <sys/hypervisor.h>
 #endif
 
+boolean_t kernel_preempt_enable = B_FALSE;
```

Comment:
So, I took a look at this (with my apologies for being out); I think that the explicit `kpreempt()` when reaping is great (though please update [illumos#7232](https://www.illumos.org/issues/7232) to reflect that it is now on by default), but I'm not so sure about the `sys_rtt_kpreempt_enable` change. First, I don't like off-by-default behavior that is so fundamental; if someone is making a change to scheduling behavior, do they test with this on or not?  If so, they may well discover failure modes that are not currently known (it is, after all, a potentially significant change); how should this affect their work?  There are some hygiene issues as well (i.e., that this is architecture-specific but the change has only been made to x86).  If this change is to be made -- especially given that it is off-by-default -- it needs to be better justified and better explained in the comments.  Unless someone needs this particular aspect of the functionality (that is, unless we know that a downstream distro will be setting `sys_rtt_kpreempt_enable`) I would say that that particular aspect should be dropped.